### PR TITLE
feat: added code to deal with self signed certs

### LINKED ISSUE #8: handle self signed certificates for grpc client

### PR DESCRIPTION
**Pull request by z103cb**
Added self signed certs and fixed the grpc / http clients to support the ssl protocol using self signed certificates

Fixes #8

### CODE REVIEW

**Review by vaibhavjainwiz — APPROVED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -13,6 +14,22 @@
...
+
+WAIT_TIME_OUT = 10
+
+WAIT_TIME_OUT = 10
+
+CA_CERT_FILE = str(Path(__file__).parent / "resources/ca.pem")
+CLIENT_KEY_FILE = str(Path(__file__).parent / "resources/client-key.pem")
+CLIENT_CERT_FILE = str(Path(__file__).parent / "resources/client.pem")
+SERVER_KEY_FILE = str(Path(__file__).parent / "resources/server-key.pem")
+SERVER_CERT_FILE = str(Path(__file__).parent / "resources/server.pem")
```
Comment (dtrifiro) [inline tests/conftest.py]: These seem duplicated

Comment (z103cb) [reply tests/conftest.py]: will fix.

Comment (z103cb) [reply tests/conftest.py]: Fixed in [ee02430](https://github.com/opendatahub-io/caikit-nlp-client/pull/16/commits/ee02430be42d16ef76793be70ee9616d23268997)

File: src/caikit_nlp_client/http_client.py [hunk 2]
```diff
@@ -95,9 +115,22 @@ def generate_text_stream(self, model_id: str, text: str, **kwargs) -> list[str]:
         try:
             log.info(f"Calling generate_text_stream for '{model_id}'")
             json_input = create_json_request(model_id, text, **kwargs)
-            response = requests.post(  # FIXME: just a stub implementation
-                self.stream_api_url, json=json_input, timeout=10
-            )
+            if self.config.tls:
+                assert self.config.client_key_path
+                assert self.config.client_crt_path
```
Comment (dtrifiro) [inline src/caikit_nlp_client/http_client.py]: Do we need the `client_key_path` and `client_crt_path` for `tls`?  Don't we need these two for `mTLS` only

File: src/caikit_nlp_client/http_client.py [hunk 3]
```diff
@@ -61,7 +69,19 @@ def generate_text(self, model_id: str, text: str, **kwargs) -> str:
...
+
+                response = requests.post(
+                    self.api_url,
+                    json=json_input,
+                    timeout=10.0,
+                    verify=self.config.ca_crt_path,
+                    cert=(self.config.client_crt_path, self.config.client_key_path),
+                )
+            else:
+                response = requests.post(self.api_url, json=json_input, timeout=10.0)
```
Comment (dtrifiro) [inline src/caikit_nlp_client/http_client.py]:
```suggestion
            kwargs = {}
            
            if self.config.tls or self.config.mtls:
                kwargs['verify'] = self.config.ca_crt_path
            if self.config.mtls:
                 kwargs['cert'] = (self.config.client_crt_path, self.config.client_key_ath)

            response = requests.post(
                self.api_url,
                json=json_input,
                timeout=10.0,
                **kwargs
            )
```

File: src/caikit_nlp_client/http_client.py (hunk 2 above)
Comment (dtrifiro) [reply src/caikit_nlp_client/http_client.py]: See the suggestion below

File: tests/conftest.py [hunk 4]
```diff
@@ -47,8 +55,15 @@ def model_name():
     return available_models[0]
 
 
+# FIXME: allow for the code to generate fixtures with either insecure or secure
+# connections
 @pytest.fixture(scope="session")
-def caikit_nlp_runtime(grpc_server_port, http_server_port):
+def insecure() -> bool:
```
Comment (dtrifiro) [inline tests/conftest.py]: Waiting for the parametrized fixture

File: tests/conftest.py [hunk 5]
```diff
@@ -76,72 +91,92 @@ def caikit_nlp_runtime(grpc_server_port, http_server_port):
...
     return get_random_port()
 
 
-def channel_factory(host: str, port: int):
-    config = GrpcChannelConfig(
-        host=host,
-        port=port,
-        insecure=True,  # TODO: handle cases with secure=True with self signed certs
-    )
+def channel_factory(host: str, port: int, insecure: bool):
```
Comment (dtrifiro) [inline tests/conftest.py:126]: This will probably have to be parametrized with both tls=true and mtls=true. We'll do this in a separate PR

**Review by dtrifiro — APPROVED** [review]

File: src/caikit_nlp_client/http_client.py (hunk 3 above)
Comment (z103cb) [reply src/caikit_nlp_client/http_client.py]: Will do
